packages/api/endpoints/rules.js

```diff
@@ -111,17 +119,32 @@ async function put({ params: { name }, body }, res) {
 
   try {
     const oldRule = await model.get({ name });
+    const dbClient = await getKnexClient();
 
-    // if rule type is onetime no change is allowed unless it is a rerun
+    // If rule type is onetime no change is allowed unless it is a rerun
     if (body.action === 'rerun') {
       return models.Rule.invoke(oldRule).then(() => res.send(oldRule));
     }
 
+    let updatedRuleItem = cloneDeep(oldRule);
     // Remove all fields from the existing rule that are not supplied in body
     // since body is expected to be a replacement rule, not a partial rule
     const fieldsToDelete = Object.keys(oldRule).filter((key) => !(key in body));
-    const newRule = await model.update(oldRule, body, fieldsToDelete);
-
+    updatedRuleItem = omit(updatedRuleItem, fieldsToDelete);
+    merge(updatedRuleItem, body);
+
+    const stateChanged = body.state && body.state !== oldRule.state;
+    const valueUpdated = body.rule && body.rule.value !== oldRule.rule.value;
+    const record = await model.updateRuleTrigger(updatedRuleItem, stateChanged, valueUpdated);
```

Comment:
Can you explain why you moved this logic into the endpoint from the model? Since it's all very specific to Dynamo, it feels like it is better suited for the model code. Also `model.update` had some helpful defensiveness around ensuring that the record was valid before doing some of this record preparation. Plus it seems like we have duplication between the endpoint and model code now

Reply:
I moved this logic into the endpoint from the model because I wanted to ensure that, if a rule type is anything other than a `onetime` rule, that the rule triggers are updated too, but I see now how that is duplicate and that will be taken care of if it reaches `model.update`. I've since removed it.